### ISSUE TITLE
Make measurements API pagination more effective

### ISSUE THREAD

**Issue by fortuna** [ooni/backend#636]
The way pagination is implemented in the API is not effective: each new page is more and more expensive. It seems it scans all the previous pages before starting to return the new page. That make pagination pointless: you might as well request a very large limit. (it seems you can't really go past 500 reliably).

While offset operations are inefficient, date ranges can be skipped efficiently. Let's leverage that!

When you [generate the next_url](https://github.com/ooni/api/blob/8aa24669cc0e5aae3b31bd35a2b371ef11a97f77/newapi/ooniapi/measurements.py#L920) field, you must also update the date range.

Update `since` (if order is ascending) or `until` (if order is descending) to match the timestamp of the last returned measurement.

Notice that there may be ties in the timestamp. Measurements that were taken in the same exact second. To account for that and avoid duplicate measurements, set the `offset` to match the number of measurements with the same timestamp as the last measurement in the current page.

This will make sure you don't re-scan any previous time ranges as you move through pages, keeping the cost of each new page about constant.

The point in the code to make that change is here: https://github.com/ooni/api/blob/8aa24669cc0e5aae3b31bd35a2b371ef11a97f77/newapi/ooniapi/measurements.py#L920

**Comment by hellais** [comment]
Yes this is a good solution to the issue reported in: https://github.com/ooni/backend/issues/127.

Am going to close the original issue in favour of this one.
